### ISSUE TITLE
Nested tuple projection not working - defaulting error

### ISSUE THREAD

**Issue by dylanmc** [GaloisInc/cryptol#2]
```
x = ( 1, (2, 3))
z = x.2.2
```

yields:

```
Cryptol> (1,(2,3)).2.2
[warning] at <interactive>:1:1--1:14:
  Defaulting type parameter 'bits'
             of literal or demoted expression
             at <interactive>:1:2--1:3
  to 1

[error] at <interactive>:1:7--1:8:
  Unsolved constraint:
    fin ?j3
      arising from
      use of literal or demoted expression
      at <interactive>:1:7--1:8
  where
  ?j3 is type parameter 'bits'
         of literal or demoted expression
         at <interactive>:1:7--1:8
[error] at <interactive>:1:7--1:8:
  Unsolved constraint:
    ?j3 >= width 3
      arising from
      use of literal or demoted expression
      at <interactive>:1:7--1:8
  where
  ?j3 is type parameter 'bits'
         of literal or demoted expression
         at <interactive>:1:7--1:8
[error] at <interactive>:1:5--1:6:
  Unsolved constraint:
    fin ?i3
      arising from
      use of literal or demoted expression
      at <interactive>:1:5--1:6
  where
  ?i3 is type parameter 'bits'
         of literal or demoted expression
         at <interactive>:1:5--1:6
[error] at <interactive>:1:5--1:6:
  Unsolved constraint:
    ?i3 >= width 2
      arising from
      use of literal or demoted expression
      at <interactive>:1:5--1:6
  where
  ?i3 is type parameter 'bits'
         of literal or demoted expression
         at <interactive>:1:5--1:6
```

**Comment by weaversa** [comment]
Will someone please explain why this is happening? I gather that since it's issue #2, it's hard to fix, but it would be nice to have documentation as to why it's so difficult.

**Comment by weaversa** [comment]
This statement ```(1, (2:[2], 3:[2])).2.1``` used to work, and now no longer works. So, this bug has gotten worse and no longer has a known workaround.

**Comment by yav** [comment]
I am not sure if this ticket started pre- or post- tuples being 0-indexed, but here is the current behavior.  Dylan's original example:

    x = ( 1, (2, 3))
    z = x.2.2

results in an error, which is correct, because `x` does not have field `2`---tuples are indexed from `0`, so only `0` and `1` work.   If we update it to the `0` based indexing:

    x = ( 1, (2, 3))
    z = x.1.1

Then things work as expected, and some types get defaulted.  Sean's examples as written also do not work because the indexes are too big, but if we decrement them by one, then things works as expected.

Can we close this ticket, or are there other problems to fix?

**Comment by yav** [comment]
I am not sure if this ticket started pre- or post- tuples being 0-indexed, but here is the current behavior.  Dylan's original example:

    x = ( 1, (2, 3))
    z = x.2.2

results in an error, which is correct, because `x` does not have field `2`---tuples are indexed from `0`, so only `0` and `1` work.   If we update it to the `0` based indexing:

    x = ( 1, (2, 3))
    z = x.1.1

Then things work as expected, and some types get defaulted.  Sean's examples as written also do not work because the indexes are too big, but if we decrement them by one, then things works as expected.

Can we close this ticket, or are there other problems to fix?

**Comment by brianhuffman** [comment]
It looks like this was fixed by the same commit (39b1e84) that fixed #418.

**Comment by weaversa** [comment]
Woohoo!
